code/util/BaseInteractable.cs

```diff
@@ -0,0 +1,82 @@
+﻿namespace Sandbox.util;
+
+public class CanTriggerResults
+{
+    public CanTriggerResults(bool hit, float dist, BaseInteractable interactable, Vector3 hitpos)
+    {
+        Hit = hit;
+        Distance = dist;
+        Interactable = interactable;
+        HitPosition= hitpos;
+    }
+
+    public bool Hit;
+    public float Distance;
+    public BaseInteractable Interactable;
+    public Vector3 HitPosition;
+}
+
+public partial class BaseInteractable : BaseNetworkable
+{
+    [Net]
+    public Entity Parent { get; set; }
+    [Net]
+    public Vector3 Mins { get; set; }
+    [Net]
+    public Vector3 Maxs { get; set; }
+    public float MaxDistance { get; set; } = 60f;
+    public CanTriggerResults LastTriggerResults {get; set; }
```

Comment:
The way you are using this is dangerous and I think indicates an anti-pattern. This should be removed.